akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedSetup.scala

```diff
@@ -61,5 +61,24 @@ private[persistence] final case class EventsourcedSetup[C, E, S](
 
   def selfUntyped = context.self.toUntyped
 
+  private var mdc: Map[String, Any] = Map.empty
+  private var _log = context.log // changed when mdc is changed
+  def log: Logger = {
+    if (_log == null)
```

Comment:
`OptionVal` > `null`

Also, is this optimization really needed, putting a lazy var depending on another var inside a case class smells a bit to me. How many transitions/new mdc decorators was created before?
Could we do the same optimization in some other less imperative way? Just add a field to the case class?

Reply:
OptionVal no prob.

We switch mdc for each persisted event. Logging is rare so creating new Logger for each switch felt wasteful.

Copy of setup for each switch also felt wasteful and would introduce a var in other place. Setup is anyway not fully immutable.

Reply:
Hmm. ok, fair enough. Thanks for the motivation.

Reply:
Should we maybe just make it a regular class altogether then? Avoiding any copy at all...

Reply:
Doesn't necessarily have to be done in this PR though.

Reply:
changed